### ISSUE TITLE
AI Tiered Plans: upgrade prompt may show up after the upgrade already happened

### ISSUE THREAD

**Issue by lhkowalski** [Automattic/jetpack#34467]
- the UI showed "20 / 100 requests", so we are sure the plan was there
- may be related to current plan caching, we could reduce from 1 minute to 10 seconds

![Image](https://github.com/Automattic/jetpack/assets/6760046/7ad72391-8d16-40da-8923-40c8787808f3)

**Comment by lhkowalski** [comment]
This is actually a very strange behavior, because the upgrade prompt being showed is not for the tiers, but for the old free + unlimited paid plan structure. I will try to replicate the issue while also revisit the caching duration.

**Comment by lhkowalski** [comment]
I was able to test the impact of the plan caching on this, and it resulted on a more consistent problem: I keep seeing the free plan, and then it switches completely to the new plan after buying it. No mixed results. So far I was not able to reproduce the same issue.

**Comment by lhkowalski** [comment]
I was not able to reproduce the issue. My guess is that it could be related to the last changes we did on the code while the the call for testing was running, maybe a problematic piece that got solved on the same day or the day after (this is coming from the first tester, and no one reported the same issue after the first one).

I am going to close this issue and observe any behavior that could be similar to this.
